Code/Legacy/CrySystem/Log.cpp

```diff
@@ -69,6 +75,58 @@ namespace LogCVars
 static CLog::LogStringType indentString ("    ");
 #endif
 
+// Gruber patch begin // AE -- update log while waiting for assets
+#if defined(CARBONATED)
+class LogNotificationProcessor
+    : public AZ::LogNotification::LogNotificatorBus::Handler
+{
+public:
+    LogNotificationProcessor()
+        : m_log(nullptr)
+    {
+    }
+
+    void SetLog(CLog* log)
+    {
+        if (m_log == nullptr && log != nullptr)
+        {
+            m_log = log;
+            AZ::LogNotification::LogNotificatorBus::Handler::BusConnect();
+        }
+        else if (m_log != nullptr && log == nullptr)
+        {
+            AZ::LogNotification::LogNotificatorBus::Handler::BusDisconnect();
+            m_log = log;
+        }
+        else
+        {
+            m_log = log;
+        }
+    }
+
+    // LogNotificatorBus imnplementation
+
+    void Update() override
+    {
+        if (m_log)
+        {
+            m_log->Update();
+        }
+    }
+
+    static LogNotificationProcessor* GetInstance()
+    {
+        static LogNotificationProcessor processor;
```

Comment:
It could create a copy of a static instance object  in each compiled library where this header will be used.
@alexeykostin ^^^^ isn't it?

Reply:
Yes, I keep that in mind. Log.cpp is part of CrySystem, which is a separate dll (build\windows\bin\profile\CrySystem.dll).
Unlike AzCore which is linked to every module.